src/armory/SelectTank.js

```diff
@@ -5,32 +5,69 @@ import Tank from '../tanks/Tank.js';
 
 type Props = {|
 	allTanks: Array<Tank>,
-	showTanks: boolean,
-	changeSelectedTank: (string) => void,
+	changeSelectedTank: (Tank) => void,
 |};
 
-type State = {||};
+type State = {|
+	showTanks: boolean,
+	selectedTank: ?Tank
+|};
 
 class SelectTank extends React.Component<Props, State> {
 
+	constructor(props: Props) {
+		super(props);
+		this.state = {
+			showTanks: false,
+			selectedTank: props.allTanks[0],
+		};
+	}
+
+	onChangeSelectedTank(selectedTank: Tank): void {
+		this.props.changeSelectedTank(selectedTank);
+		this.setState({
+			selectedTank: selectedTank,
+			showTanks: false
+		});
+	}
+
+	componentDidUpdate(prevProps: Props) {
+		if (this.props !== prevProps) {
+			this.setState({selectedTank: this.props.allTanks[0]});

```

Comment:
By default, we get the most recent created tank.

Should be `this.setState({selectedTank: this.props.allTanks[this.props.allTanks.length - 1]});